lessons/python/inttopics/inttopics.ipynb

```diff
@@ -0,0 +1,672 @@
+    "\n",
+    "# conditionals can also be included in list comprehensions - this is valid syntax\n",
+    "\n",
+    "scores = [int(num)*0.4 for num in scores if type(num) is str] "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "For large operations, list comprehensions are also faster than using `list.append`. When the latter is a part of a loop, Python looks up the list in memory at every single iteration, which can really slow things down for large lists. List comprehensions, on the other hand, do not require the look up operation. They're also far more visually compact while conveying the same amount of information."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "print(scores)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "len(scores) "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Notice how the value we appended is gone - this is because the list comprehension is assigned to `scores` above, and actually overwrites it. So be careful with list comprehensions, since they don't just perform a specified operation while leaving everything else intact - on their own, they 'become' a list in their own right!\n",
+    "\n",
+    "For future reference, we could use an assertion to make sure we did things right. We'll talk about that further down."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "# putting it all together - the right way, all along!\n",
+    "\n",
+    "with open('scores.txt','r') as f:\n",
+    "     scores = [int(num) for num in f.read().split(',')] "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "print(scores) "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Control Flow Tools - Pass, Continue, Break"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "So you'll notice that in that sample loop before the list comprehension, we used a 'pass' statement. Here it is again:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "for num in scores:\n",
+    "    if type(num) is str:\n",
+    "        num = int(num)\n",
+    "        num = num * 0.4\n",
+    "    else:\n",
+    "        pass"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This is part of a family of useful operators within Python that allow for more precise loop operations."
```

Comment:
That's a fair point - I was indeed referring to `pass` with 'this', but I'll clarify that it's not restricted to just loops.